### ISSUE TITLE
Timeouts with elm-format/ElmLS

### ISSUE THREAD

**Issue by jc00ke** [monty5811/postcss-elm-tailwind#6]
HI there, this is awesome! I'm using it on a side project and I'm unable to use `elm-format` via [elm-language-server](https://github.com/elm-tooling/elm-language-server) because (I'm speculating, but I'm pretty sure) the vendored `TW.elm` file is 56,627 lines long. Any ideas how one might still use this file with the LS and `elm-format` via `coc.nvim`?

**Comment by monty5811** [comment]
I'm not sure... 

I tend to not open the generated file in an editor that tries to do anything fancy, because it's so large...

I run `elm-format` using a makefile & I have a rule to delete the generated `TW.elm` before running `elm-format`. Not the best solution, but it works...

There might be a way to exclude files by name when using the LS or `coc.nvim`? If not, maybe worth a feature request on those projects?

**Comment by jc00ke** [comment]
Thanks, I'll have to look at my options with `coc.nvim` and the `LS`. In the meantime, I'm just grabbing the functions I need (luckily not many).

**Comment by yonigibbs** [comment]
I just hit this same issue. I had an npm script which runs `elm-format --validate` on my source folder during a build, and it was hanging. I worked around it by changing the npm script to instead call a custom shell script which loops round the files in the folder and calls `elm-format --validate` on each one, apart from `TW.elm`. Putting it below in case it's of use to anyone. (It doesn't handle a nested folder structure as I don't currently have that, but would be trivial to add that.)

As a longer-term solution, though, do you think it might be worth logging an issue against `elm-format` to allow it to exclude certain files (typically generated files) based on some glob or something? Let me know if you agree and I'll log one.

```
set -e

for file in src/*.elm; do
  if [ "$file" != "src/TW.elm" ]; then
    npx elm-format --validate "$file"
  fi
done
```

**Comment by jc00ke** [comment]
@yonigibbs give `elm-format` 0.8.3 a try. I tried it last night and it worked much better.
https://github.com/avh4/elm-format/releases/tag/0.8.3

I also manually changed the `exposing ( every, function... )` to `exposing (..)`. `elm-format` will leave that be now.

**Comment by yonigibbs** [comment]
Woo. That's a **big** difference in the new release!  It now manages to process the file in a few seconds.  I noticed that the auto-generated `TW.elm` actually fails the validation from `elm-format`, so @monty5811, maybe that's a change that could be applied to this project here, to make it generate code which complies with `elm-format`?

But anyway it's not a big deal, because if you just run `elm-format` on the file once, it sorts it out, and then `elm-format --validate` works. Just means you need to remember to do it every time `TW.elm` is regenerated (or manually exclude it from the `validate` step).

**Comment by monty5811** [comment]
That's a great perf improvement!

I'll modify the output to pass `elm-format`. Easier to test now I can run `elm-format` on `TW.elm` in a timely fashion.

**Comment by yonigibbs** [comment]
Great stuff, thanks @monty5811.  I've also logged an issue requesting the ability to exclude files: https://github.com/avh4/elm-format/issues/669. Even though the performance is hugely improved with this new version, it's still taking a few seconds to process `TW.elm` so it means that validating your whole source folder goes from a few tenths of a second to several seconds.

**Comment by monty5811** [comment]
`elm-format` compatible output released in `0.6.0`
